src/sentry/web/frontend/auth_login.py

```diff
@@ -67,11 +69,168 @@ def run_callbacks(self, request: Request):
-                slug=organization_slug, status=OrganizationStatus.ACTIVE
+    @never_cache
+    def handle(self, request: Request, *args, **kwargs) -> HttpResponse:
+        """
+        Hooks in to the django view dispatch which delegates request to GET/POST/PUT/DELETE.
+        Base view overwrites dispatch to include functionality for csrf, superuser, customer domains, etc.
+        """
+        return super().handle(request=request, *args, **kwargs)
+
+    def get(self, request: Request, **kwargs) -> HttpResponse:
+        next_uri = self.get_next_uri(request=request)
+        if request.user.is_authenticated:
+            return self.redirect_authenticated_user(request=request, next_uri=next_uri)
+
+        request.session.set_test_cookie()
+
+        # Initiate login clears the pending session states so you don't end up at an odd location
+        initiate_login(request=request, next_url=next_uri)
+
+        # Single org mode -- send them to the org-specific handler
+        if settings.SENTRY_SINGLE_ORGANIZATION:
+            return self.redirect_to_org_specific_login()
+
+        session_expired = "session_expired" in request.COOKIES
+        if session_expired:
+            messages.add_message(
+                request=request, level=messages.WARNING, message=WARN_SESSION_EXPIRED
+            )
+
+        default_context = self.get_default_context(request=request)
+
+        if self.should_redirect_to_sso_login(request=request):
+            response = self.get_sso_redirect(request=request)
+        elif self.can_register(request=request):
+            response = self.get_registration_page(
+                request=request, context=default_context, **kwargs
+            )
+        else:
+            response = self.get_login_page(request=request, context=default_context, **kwargs)
+
+        if session_expired:
+            response.delete_cookie("session_expired")
+
+        return response
+
+    def get_next_uri(self, request: Request) -> str:
+        """
+        Returns the next URI a user should visit in their authentication flow.
+        """
+        next_uri_fallback = None
+        if request.session.get("_next") is not None:
+            next_uri_fallback = request.session.pop("_next")
+        return request.GET.get(REDIRECT_FIELD_NAME, next_uri_fallback)
+
+    def redirect_authenticated_user(self, request: Request, next_uri: str) -> HttpResponseRedirect:
+        """
+        If an authenticated user sends a GET request to AuthLoginView, we redirect them forwards in the auth process.
+        """
+        if is_valid_redirect(url=next_uri, allowed_hosts=(request.get_host(),)):
+            return self.redirect(url=next_uri)
+        return self.redirect_to_org(request=request)
+
+    def redirect_to_org_specific_login(self) -> HttpResponseRedirect:
+        """
+        If Sentry is in single org mode this redirects users to their specific handler.
+        """
+        org = Organization.get_default()
```

Comment:
Queries to `Organization` should be done via `organization_service` as this view is not guaranteed to be called exclusively in the region silo.

```suggestion
        org = organization_service.get_default_organization()
```

Reply:
Does this include things like `Organization.get()`? The previous code wasn't using this and I only see it being used in one place in the codebase - would it be better to swap that over in a follow-up PR in order to avoid scope-creep/any risk from making an actual change, or should we put it in here?